File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    # 개수 기준 내림차순 정렬
    sorted_tangerine = sorted(Counter(tangerine).items(), key=lambda x:x[1], reverse=True)

    types = 0
    nums = 0
    
    for t, n in sorted_tangerine:
        nums += n
        types += 1
        
        if nums >= k:
            return types

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var sum = 0;
    var res = 0;
    var t_num = [];
    for (const t of tangerine) {
        t_num[t] = t_num[t] ? t_num[t] + 1 : 1;
    }
    
    t_num.sort((a, b) => b - a);
    
    for (const n of t_num) {
        sum += n;
        res += 1;
        if (sum >= k) return res;
    }
}